python/cudf_polars/cudf_polars/dsl/tracing.py

```diff
@@ -177,7 +180,15 @@ def wrapper(
                 device_handle=maybe_handle,
                 pid=pid,
             )
-            record = before | after
+            after_end = time.monotonic_ns()
+            record = (
+                before
+                | after
+                | {
+                    "overhead_duration": (before_end - before_start)
+                    + (after_end - after_start)
```

Comment:
suggestion: If you'd ever want to distinguish the input phase overhead from the output phase overhead, you could store `before_end - before_start` in a separate field from `after_end - after_start`, but maybe that level of detail is not needed here

Reply:
> but maybe that level of detail is not needed here

That was my thinking. I suspect we won't need that level of granularity right now.